### PR TITLE
simple gui implementation in javafx

### PR DESCRIPTION
**Pull request by tomeks86**

### CODE REVIEW

File: client/src/main/java/com/battleships/Client.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import javafx.fxml.FXMLLoader;
+import javafx.scene.Parent;
+import javafx.scene.Scene;
+import javafx.stage.Stage;
+
+public class Client extends Application {
+
+    @Override
+    public void start(Stage primaryStage) throws Exception {
+        System.out.println(getClass().getResource("start_window/start_window.fxml").toString());
```
Comment (AdrianHarenczyk) [inline client/src/main/java/com/battleships/Client.java]: Sądzę, że nie powinno być tego elementu w tym miejscu. :)

File: client/src/main/java/com/battleships/Client.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        FXMLLoader loader = new FXMLLoader(getClass().getResource("start_window/start_window.fxml"));
+        Parent root = loader.load();
+        primaryStage.setTitle("battleships");
+        primaryStage.setScene(new Scene(root, 800, 600));
```
Comment (boczkas) [inline client/src/main/java/com/battleships/Client.java]: Maybe get system resolution somehow?

File: client/src/main/java/com/battleships/Translator.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+public class Translator {
+    private static final ObjectProperty<Locale> locale;
+
+    static {
+        locale = new SimpleObjectProperty<>(Locale.getDefault());
+        locale.addListener(((observable, oldValue, newValue) -> Locale.setDefault(newValue)));
+    }
+
+    public static List<Locale> getSupportedLocales() {
```
Comment (boczkas) [inline client/src/main/java/com/battleships/Translator.java]: private constructor?

**Review by AdrianHarenczyk — APPROVED** [review]
Ogólnie wygląda dobrze. Można ewentualnie zaznaczyć bardziej adekwatne informacje w Readme.md

File: client/src/main/java/com/battleships/Translator.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    public static void setLocale(Locale locale) {
+        localeProperty().set(locale);
+        Locale.setDefault(locale);
+    }
+
+    public static ObjectProperty<Locale> localeProperty() {
+        return locale;
+    }
+
+    public static String get(final String key, final Object... args) {
```
Comment (boczkas) [inline client/src/main/java/com/battleships/Translator.java]: getTranslation

File: client/src/main/java/com/battleships/Translator.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    public static StringBinding createStringBinding(final String key, Object... args) {
+        return Bindings.createStringBinding(() -> get(key, args), locale);
+    }
+
+    public static StringBinding createStringBinding(Callable<String> func) {
+        return Bindings.createStringBinding(func, locale);
+    }
+
+    public static void bind(Labeled settingsLabel, String key, Object... args) {
+        settingsLabel.textProperty().bind(createStringBinding(key, args));
```
Comment (boczkas) [inline client/src/main/java/com/battleships/Translator.java]: train

File: client/src/main/java/com/battleships/start_window/FooterControler.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+import javafx.fxml.FXML;
+import javafx.scene.control.ChoiceBox;
+
+import java.util.Locale;
+
+public class FooterControler {
+    @FXML
+    private ChoiceBox<String> languageSelector;
+
+    @FXML public void initialize(){
```
Comment (boczkas) [inline client/src/main/java/com/battleships/start_window/FooterControler.java]: TomekS:
@FXML convention

File: client/src/main/java/com/battleships/start_window/FooterControler.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+        setupLanguageSelector();
+    }
+
+    private void setupLanguageSelector() {
+        languageSelector.setItems(FXCollections.observableArrayList( "pl", "en"));
+        languageSelector.getSelectionModel().selectedItemProperty().addListener((observable, oldV, newV) -> {
```
Comment (boczkas) [inline client/src/main/java/com/battleships/start_window/FooterControler.java]: train

File: client/src/main/java/com/battleships/start_window/FooterControler.java [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+                    Translator.setLocale(new Locale(newV));
+                }
+        );
+        languageSelector.getSelectionModel().selectFirst();
```
Comment (boczkas) [inline client/src/main/java/com/battleships/start_window/FooterControler.java]: train

File: client/src/main/java/com/battleships/start_window/HeaderController.java [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+
+import com.battleships.Translator;
+import javafx.fxml.FXML;
+import javafx.scene.control.Label;
+
+public class HeaderController {
+    @FXML
+    private Label settingsLabel;
+
+    @FXML public void initialize() {
```
Comment (boczkas) [inline client/src/main/java/com/battleships/start_window/HeaderController.java]: @FXML convention

File: client/src/main/java/com/battleships/start_window/InsertData/InsertSeekBarData.java [hunk 10]
```diff
@@ -0,0 +1,40 @@
+package com.battleships.start_window.InsertData;
+
+import com.battleships.Translator;
+import javafx.fxml.FXML;
+import javafx.scene.control.Label;
+import javafx.scene.control.Slider;
+
+public class InsertSeekBarData {
```
Comment (boczkas) [inline client/src/main/java/com/battleships/start_window/InsertData/InsertSeekBarData.java]: Name of class sounds like method name
SeekBarDataInserter?

File: client/src/main/java/com/battleships/start_window/InsertData/InsertSeekBarData.java [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+
+
+    @FXML
+    private Label columnsTextLabel;
+    @FXML
+    private Slider columnsProgressBar;
+    @FXML
+    private Label columnsAmountDisplayer;
+
+    public void initialize() {
```
Comment (boczkas) [inline client/src/main/java/com/battleships/start_window/InsertData/InsertSeekBarData.java]: in FooterController initialize is annotated with @FXML, here it is not

File: client/src/main/java/com/battleships/start_window/InsertData/InsertSeekBarData.java [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+        initRowsProgressBar();
+        initColumnsProgressBar();
+    }
+
+    private void initRowsProgressBar() {
+        Translator.bind(rowsTextLabel, "row_count");
+        rowsProgressBar.valueProperty().addListener
+                ((observableValue, oldValue, newValue) -> rowsAmountDisplayer.setText(String.valueOf(newValue.intValue())));
```
Comment (boczkas) [inline client/src/main/java/com/battleships/start_window/InsertData/InsertSeekBarData.java]: train

File: client/src/main/java/com/battleships/start_window/InsertData/InsertTextData.java [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+
+public class InsertTextData {
+    @FXML
+    private Button connectToServer;
+    @FXML
+    private TextField nameTextField;
+
+    @FXML public void initialize(){
+        Translator.bind(connectToServer, "connect");
+        nameTextField.promptTextProperty().bind(Translator.createStringBinding("player_name"));
```
Comment (boczkas) [inline client/src/main/java/com/battleships/start_window/InsertData/InsertTextData.java]: train

File: client/src/main/java/com/battleships/start_window/RoomListController.java [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+import com.battleships.Translator;
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+
+public class RoomListController {
+    @FXML
+    private Label roomListLabel;
+    @FXML
+    private Button createRoom;
```
Comment (boczkas) [inline client/src/main/java/com/battleships/start_window/RoomListController.java]: createRoomButton?

File: client/src/main/java/com/battleships/start_window/RoomListController.java [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+    @FXML
+    private Button selectRoom;
```
Comment (boczkas) [inline client/src/main/java/com/battleships/start_window/RoomListController.java]: selectRoomButton

File: client/src/main/resources/translations_en.properties [hunk 16]
```diff
@@ -0,0 +1,8 @@
+choose_settings=Choose settings
+select_room=Select room
+create_room=Create room
+connect=Connect
+room_list=Rooms list
+player_name=Player name
+row_count=Row count
```
Comment (boczkas) [inline client/src/main/resources/translations_en.properties]: plural / singular in row/columns texts - decide

**Review by boczkas — CHANGES_REQUESTED** [review]
Some cosmetical changes needed

File: client/src/main/java/com/battleships/Client.java (hunk 1 above)
Comment (dmuca) [reply client/src/main/java/com/battleships/Client.java]: Usunięte

File: client/src/main/java/com/battleships/Translator.java (hunk 4 above)
Comment (dmuca) [reply client/src/main/java/com/battleships/Translator.java]: changed

File: client/src/main/java/com/battleships/Translator.java (hunk 5 above)
Comment (dmuca) [reply client/src/main/java/com/battleships/Translator.java]: fixed

File: client/src/main/java/com/battleships/start_window/FooterControler.java (hunk 6 above)
Comment (dmuca) [reply client/src/main/java/com/battleships/start_window/FooterControler.java]: fixed

File: client/src/main/java/com/battleships/start_window/FooterControler.java (hunk 7 above)
Comment (dmuca) [reply client/src/main/java/com/battleships/start_window/FooterControler.java]: fixed

File: client/src/main/java/com/battleships/start_window/FooterControler.java (hunk 8 above)
Comment (dmuca) [reply client/src/main/java/com/battleships/start_window/FooterControler.java]: fixed

File: client/src/main/java/com/battleships/start_window/InsertData/InsertSeekBarData.java (hunk 10 above)
Comment (dmuca) [reply client/src/main/java/com/battleships/start_window/InsertData/InsertSeekBarData.java]: done

File: client/src/main/java/com/battleships/start_window/InsertData/InsertSeekBarData.java (hunk 11 above)
Comment (dmuca) [reply client/src/main/java/com/battleships/start_window/InsertData/InsertSeekBarData.java]: done

File: client/src/main/java/com/battleships/start_window/RoomListController.java (hunk 14 above)
Comment (dmuca) [reply client/src/main/java/com/battleships/start_window/RoomListController.java]: done

File: client/src/main/java/com/battleships/start_window/RoomListController.java (hunk 15 above)
Comment (dmuca) [reply client/src/main/java/com/battleships/start_window/RoomListController.java]: done

File: client/src/main/resources/translations_en.properties (hunk 16 above)
Comment (dmuca) [reply client/src/main/resources/translations_en.properties]: changed

File: client/src/main/java/com/battleships/Client.java (hunk 2 above)
Comment (dmuca) [reply client/src/main/java/com/battleships/Client.java]: todo later
https://github.com/JavaAcademyDoktorki/battleships/issues/6

File: client/src/main/java/com/battleships/start_window/HeaderController.java (hunk 9 above)
Comment (dmuca) [reply client/src/main/java/com/battleships/start_window/HeaderController.java]: fixed

File: client/src/main/java/com/battleships/start_window/InsertData/InsertSeekBarData.java (hunk 12 above)
Comment (dmuca) [reply client/src/main/java/com/battleships/start_window/InsertData/InsertSeekBarData.java]: it's a GUI, train allowed

File: client/src/main/java/com/battleships/start_window/InsertData/InsertTextData.java (hunk 13 above)
Comment (dmuca) [reply client/src/main/java/com/battleships/start_window/InsertData/InsertTextData.java]: it's a GUI, train allowed

File: client/src/main/java/com/battleships/Translator.java [hunk 17]
```diff
@@ -18,44 +19,29 @@
         locale.addListener(((observable, oldValue, newValue) -> Locale.setDefault(newValue)));
```
Comment (boczkas) [inline client/src/main/java/com/battleships/Translator.java]: to many brackets

**Review by boczkas — APPROVED** [review]
Okejka

**Pull request merged**